pallets/liquid-staking/src/lib.rs

```diff
@@ -917,14 +971,19 @@ pub mod pallet {
                 return Ok(());
             }
 
+            if StakingLedgers::<T>::contains_key(&derivative_index) {
+                return Self::do_bond_extra(derivative_index, amount);
+            }
+
             ensure!(
                 T::DerivativeIndexList::get().contains(&derivative_index),
                 Error::<T>::InvalidDerivativeIndex
             );
-            ensure!(
-                !StakingLedgers::<T>::contains_key(&derivative_index),
-                Error::<T>::AlreadyBonded
-            );
+            // ensure!(
+            //     amount >= T::MinNominatorBond::get(),
```

Comment:
we still need this ensure to do `assert` job in case we have an error in our logic